packages/varlet-ui/src/date-picker/__tests__/index.spec.js

```diff
@@ -422,3 +422,27 @@ test('test datePicker titleColor', async () => {
 
   wrapper.unmount()
 })
+
+test('test datePicker reset when modelValue is undefined or an empty string', async () => {
+  const wrapper = mount({
+    components: {
+      [VarDatePicker.name]: VarDatePicker,
+    },
+    data() {
+      return {
+        date: '2024-09-23',
+      }
+    },
+    template: `<var-date-picker v-model="date" />`,
+  })
+
+  expect(wrapper.html()).toMatchSnapshot()
+
+  await wrapper.setData({ time: undefined })
```

Comment:
The data you set is `date`, not `time`.  Adding snapshot after `setData` may be a good practice to check whether the time is right.